modin/test/test_partition_api.py

```diff
@@ -54,27 +54,50 @@
 
 
 @pytest.mark.parametrize("axis", [None, 0, 1])
-def test_unwrap_partitions(axis):
+@pytest.mark.parametrize("reset_index", [True, False])
+@pytest.mark.parametrize("reset_columns", [True, False])
+def test_unwrap_partitions(axis, reset_index, reset_columns):
     data = test_data["int_data"]
-    df = pd.DataFrame(data)
 
+    # Create the dataframe twice-- once to get expected partitions,
+    # and another time to get the partition with unwrap_partitions.
+    # Both the test and the code under test should mutate the dataframe
+    # by calling _propagate_index_objs, so we should use two separate
+    # dataframes.
+    def get_df(data):
+        df = pd.DataFrame(data)
+        if reset_index:
+            df.index = df.index[::-1]
+        if reset_columns:
+            df.columns = df.columns[::-1]
+        return df
+
+    df = get_df(data)
+    # Propagate any deferred updates to the partitions and drain call
+    # queues before getting the data.
+    expected_df = get_df(data)
+    expected_df._query_compiler._modin_frame._propagate_index_objs(None)
+    [
+        p.drain_call_queue()
+        for p in expected_df._query_compiler._modin_frame._partitions.flatten()
+    ]
+    expected_partitions = expected_df._query_compiler._modin_frame._partitions
```

Comment:
Maybe we should better create this "reversed" df by building then reversing it in pandas and then converting to Modin? For now this test seems to be repeating what is done in the code, what is the point of checking that?

Reply:
Done. I agree that that's better. I was uncomfortable doing the same thing in the unit test and the code, too.